Fix Farmer Reporting Missing Equipment After Completing All Field Work

rivon0507 opened the pull request:
Closes #10911 

# Changes proposed in this pull request
- Having no more fields to work on does not mean the farmer is idle at their job : they're not missing anything, just done with the day's field work
- logging has been introduced to capture cases where the AI encounters unexpected extensions. This should help in monitoring potential data inconsistencies that may not yet cause failures but are still worth tracking

## Testing
- [x] Yes I tested this before submitting it.
- [ ] I also did a multiplayer test.

Review please

rivon0507 committed:
Add check for null field and log unexpected extensions in farmer AI

`fieldToWork` being `null` means the farmer has no more field work left for the day, so they can stay idle listening for crafting requests.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=10956) <br/>All committers have signed the CLA.

rivon0507 committed:
Make the farmer `idle at job` only when it's confirmed that they are missing items or fields.

Farmers no longer start each farming preparation with `idleAtJob` set to true. Instead, this flag is only enabled after it's confirmed that the farmer lacks required items or access to a field.

rivon0507 committed:
Formatting
